ledger/complete/checkpointqueue.go

```diff
@@ -0,0 +1,87 @@
+package complete
+
+import (
+	"github.com/onflow/flow-go/ledger/complete/mtrie/trie"
+)
+
+// CheckpointQueue is a fix-sized FIFO queue of MTrie.
+// CheckpointQueue is intentionally not threadsafe given its limited use case.
+// CheckpointQueue is not a general purpose queue to avoid incurring overhead
+// for features not needed for its limited use case.
+type CheckpointQueue struct {
+	ts       []*trie.MTrie
+	capacity int
+	tail     int // element index to write to
+	count    int // number of elements (count <= capacity)
+}
+
+// NewCheckpointQueue returns a new CheckpointQueue with given capacity.
+func NewCheckpointQueue(capacity uint) *CheckpointQueue {
+	return &CheckpointQueue{
+		ts:       make([]*trie.MTrie, capacity),
+		capacity: int(capacity),
```

Comment:
> any reason to keep capacity as int?

Yes.

`capacity` function argument as `uint` is to avoid checking for negative value.  

`CheckpointQueue.capacity` as `int` is to prevent casting it to `int` when using it with `len(ts)` or slice index. 

Also, overflow when casting from `uint` to `int` is unlikely given we're only using 500 as default.